pkg/prometheus/promcfg.go

```diff
@@ -3052,7 +3052,25 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 				})
 			}
 
-			configs[i] = cg.WithMinimumVersion("2.3.0").addFiltersToYaml(configs[i], config.Filters)
+			configs[i] = cg.addFiltersToYaml(configs[i], config.Filters)
+
+			if config.FollowRedirects != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "follow_redirects",
+					Value: config.FollowRedirects,
+				})
+			}
+
+			if config.EnableHTTP2 != nil {
```

Comment:
Same here (Prometheus >= "2.35.0")

Reply:
I have added these checks in `resource_selector.go`